### PR TITLE
fix(api): `vim.cmd.make` crashes when argument count isn't 1

### LINKED ISSUE #19696: vim.cmd.make causes neovim to crash
### Neovim version (nvim -v)

v0.8.0-dev-804-gbebfbfab3b

### Vim (not Nvim) behaves the same?

n/a

### Operating system/version

Arch Linux

### Terminal name/version

foot

### $TERM environment variable

foot

### Installation

AUR

### How to reproduce the issue

```
nvim --clean
:lua vim.cmd.make()
```

### Expected behavior

Neovim runs makeprg like `:make`

### Actual behavior

Neovim crashes with the following message:
`/home/main-builder/pkgwork/src/neovim-git/src/nvim/api/command.c:822: build_cmdline_str: Assertion argc == 1 failed.`

Backtrace:
```
(gdb) bt
#0  0x00007fca5beb44dc in ?? () from /usr/lib/libc.so.6
#1  0x00007fca5be64998 in raise () from /usr/lib/libc.so.6
#2  0x00007fca5be4e53d in abort () from /usr/lib/libc.so.6
#3  0x00007fca5be4e45c in ?? () from /usr/lib/libc.so.6
#4  0x00007fca5be5d4c6 in __assert_fail () from /usr/lib/libc.so.6
#5  0x00005558343ebc3b in build_cmdline_str (cmdlinep=cmdlinep@entry=0x7ffdd84c7530, eap=eap@entry=0x7ffdd84c75a0, cmdinfo=cmdinfo@entry=0x7ffdd84c78c0, args=..., argc=argc@entry=0)
    at /home/main-builder/pkgwork/src/neovim-git/src/nvim/api/command.c:822
#6  0x00005558343ecdf7 in nvim_cmd (channel_id=9223372036854775809, cmd=<optimized out>, opts=<optimized out>, err=0x7ffdd84c7a10) at /home/main-builder/pkgwork/src/neovim-git/src/nvim/api/command.c:622
#7  0x00005558343d7f8f in nlua_api_nvim_cmd (lstate=0x7fca5c26d380) at /home/main-builder/pkgwork/src/build/src/nvim/auto/lua_api_c_bindings.generated.c:1469
#8  0x00007fca5c019ef6 in ?? () from /usr/lib/libluajit-5.1.so.2
#9  0x00007fca5c02c843 in lua_pcall () from /usr/lib/libluajit-5.1.so.2
#10 0x00005558344edb90 in nlua_pcall (lstate=0x7fca5c26d380, nargs=0, nresults=0) at /home/main-builder/pkgwork/src/neovim-git/src/nvim/lua/executor.c:134
#11 0x00005558344fb002 in nlua_typval_exec (lcmd=lcmd@entry=0x5558361c8980 "vim.cmd.make()", lcmd_len=14, name=name@entry=0x5558346686e6 ":lua", args=args@entry=0x0, argcount=argcount@entry=0, special=special@entry=false, ret_tv=0x0)
    at /home/main-builder/pkgwork/src/neovim-git/src/nvim/lua/executor.c:1304
#12 0x00005558344fb851 in ex_lua (eap=<optimized out>) at /home/main-builder/pkgwork/src/neovim-git/src/nvim/lua/executor.c:1497
#13 0x000055583449d336 in execute_cmd0 (retv=retv@entry=0x7ffdd84c7d94, eap=eap@entry=0x7ffdd84c7da0, errormsg=errormsg@entry=0x7ffdd84c7d98, preview=preview@entry=false)
    at /home/main-builder/pkgwork/src/neovim-git/src/nvim/ex_docmd.c:1635
#14 0x000055583449f099 in do_one_cmd (cmdlinep=cmdlinep@entry=0x7ffdd84c7ff8, flags=flags@entry=0, cstack=cstack@entry=0x7ffdd84c8080, fgetline=fgetline@entry=0x5558344ac9e2 <getexline>, cookie=cookie@entry=0x0)
    at /home/main-builder/pkgwork/src/neovim-git/src/nvim/ex_docmd.c:2284
#15 0x000055583449fb8f in do_cmdline (cmdline=<optimized out>, fgetline=0x5558344ac9e2 <getexline>, cookie=0x0, flags=0) at /home/main-builder/pkgwork/src/neovim-git/src/nvim/ex_docmd.c:594
#16 0x0000555834542c77 in nv_colon (cap=0x7ffdd84c86b0) at /home/main-builder/pkgwork/src/neovim-git/src/nvim/normal.c:4044
#17 0x000055583453feda in normal_execute (state=0x7ffdd84c8630, key=<optimized out>) at /home/main-builder/pkgwork/src/neovim-git/src/nvim/normal.c:1165
#18 0x00005558345f0bd4 in state_enter (s=0x7ffdd84c8630) at /home/main-builder/pkgwork/src/neovim-git/src/nvim/state.c:88
#19 0x000055583453c8e4 in normal_enter (cmdwin=<optimized out>, noexmode=<optimized out>) at /home/main-builder/pkgwork/src/neovim-git/src/nvim/normal.c:464
#20 0x0000555834503035 in main (argc=<optimized out>, argv=<optimized out>) at /home/main-builder/pkgwork/src/neovim-git/src/nvim/main.c:581
```

### PR DESCRIPTION
**Pull request by famiu**
Closes #19696

### CODE REVIEW

**Comment by zeertzjq** [comment]
This still crashes if you pass two arguments to `vim.cmd.make`

**Comment by famiu** [comment]
> This still crashes if you pass two arguments to `vim.cmd.make`



> This still crashes if you pass two arguments to `vim.cmd.make`

should be fixed now

File: test/functional/api/vim_spec.lua [hunk 1]
```diff
@@ -3829,5 +3829,11 @@ describe('API', function()
       eq({'aa'}, meths.buf_get_lines(0, 0, 1, false))
       assert_alive()
     end)
+    it("'make' command works when argument count isn't 1 #19696", function()
+      meths.cmd({ cmd = 'make' }, {})
+      assert_alive()
+      meths.cmd({ cmd = 'make', args = { 'foo', 'bar' } }, {})
+      assert_alive()
```
Comment (zeertzjq) [inline test/functional/api/vim_spec.lua:3836]: This test can potentially be dangerous.

Comment (famiu) [reply test/functional/api/vim_spec.lua:3836]: Hmm... is there any way to bypass that? I guess we can just remove the test if there's no scope for a regression?

Comment (clason) [reply test/functional/api/vim_spec.lua:3836]: needs to set `makeprg` to something innocuous?

Comment (famiu) [reply test/functional/api/vim_spec.lua:3836]: Oh I guess that's a way, now I feel stupid for not thinking of that

Comment (famiu) [reply test/functional/api/vim_spec.lua:3836]: Actually there's not really anything I can think of that we can make `makeprg` use honestly... Might just be better to remove the test

**Pull request merged**